tests/scalers_go/new_relic/new_relic_test.go

```diff
@@ -0,0 +1,298 @@
+	scaledObjectName        = fmt.Sprintf("%s-so", testName)
+	secretName              = fmt.Sprintf("%s-secret", testName)
+	triggerAuthName         = fmt.Sprintf("%s-ta", testName)
+	newRelicAPIKey          = os.Getenv("NEWRELIC_API_KEY")
+	newRelicLicenseKey      = os.Getenv("NEWRELIC_LICENSE")
+	newRelicRegion          = os.Getenv("NEWRELIC_REGION")
+	newRelicAccountID       = os.Getenv("NEWRELIC_ACCOUNT_ID")
+	kuberneteClusterName    = "keda-new-relic-cluster"
+	newRelicHelmRepoURL     = "https://helm-charts.newrelic.com"
+	deploymentReplicas      = 1
+	minReplicaCount         = 0
+	maxReplicaCount         = 2
+)
+
+type templateData struct {
+	TestNamespace           string
+	DeploymentName          string
+	MonitoredDeploymentName string
+	ServiceName             string
+	ScaledObjectName        string
+	TriggerAuthName         string
+	SecretName              string
+	NewRelicAPIKey          string
+	DeploymentReplicas      string
+	NewRelicRegion          string
+	NewRelicAccountID       string
+	KuberneteClusterName    string
+	MinReplicaCount         string
+	MaxReplicaCount         string
+}
+type templateValues map[string]string
+
+const (
+	secretTemplate = `apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+data:
+  newRelicApiKey: {{.NewRelicAPIKey}}
+`
+
+	triggerAuthenticationTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthName}}
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+  - parameter: queryKey           # Required.
+    name: {{.SecretName}}         # Required.
+    key: newRelicApiKey           # Required.
+`
+
+	monitoredDeploymentTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.MonitoredDeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.MonitoredDeploymentName}}
+spec:
+  replicas: {{.DeploymentReplicas}}
+  selector:
+    matchLabels:
+      app: {{.MonitoredDeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.MonitoredDeploymentName}}
+    spec:
+      containers:
+      - name: prom-test-app
+        image: tbickford/simple-web-app-prometheus:a13ade9
+        imagePullPolicy: IfNotPresent
+`
+
+	deploymentTemplate = `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{.DeploymentName}}
+  namespace: {{.TestNamespace}}
+  labels:
+    app: {{.DeploymentName}}
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: {{.DeploymentName}}
+  template:
+    metadata:
+      labels:
+        app: {{.DeploymentName}}
+    spec:
+      containers:
+      - name: prom-test-app
+        image: tbickford/simple-web-app-prometheus:a13ade9
+        imagePullPolicy: IfNotPresent
+`
+
+	serviceTemplate = `apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    name: {{.ServiceName}}
+    namespace: {{.TestNamespace}}
+  annotations:
+    prometheus.io/scrape: "true"
+  name: {{.ServiceName}}
```

Comment:
```suggestion
metadata:
  labels:
    name: {{.ServiceName}}
  annotations:
    prometheus.io/scrape: "true"
  name: {{.ServiceName}}
  namespace: {{.TestNamespace}}
```

Reply:
The namespace is inside the labels and that puts it in default namespace instead of the correct (and the test fails).
I have suggested the change to solve the issue

Reply:
🤦🏼 must have happened while I was moving things around (indentation). Sorry about that.